org.armagetronad.ArmagetronAdvanced.json

```diff
@@ -8,10 +8,12 @@
         "--share=ipc",
         "--socket=x11",
         "--share=network",
-        "--socket=pulseaudio"
+        "--socket=pulseaudio",
+        "--filesystem=~/.org.armagetronad.ArmagetronAdvanced/var:create",
```

Comment:
This is quite nonstandard directory. I wonder what is wrong with the current one inside `~/.var/app/org.armagetronad.ArmagetronAdvanced`? :-)

Reply:
That it didn't work properly, see #3. But yes, we have command line arguments to change those directories to a more modern place and should use them here. ~/.appname is so 2001.

Reply:
Ah, I just found #3 as well. :-) Anyway, `--persist=.org.armagetronad.ArmagetronAdvanced` (not that there is no `~/` prefix used) could be used instead. Nice thing is that it is 1. just one line instead of 3 lines for the --filesystem finish-args and 2. that it creates the `.org.armagetronad.ArmagetronAdvanced` inside the app sandboxed directory structure instead of directly in the user home directory. With `--persist`, it will be created as ~/.var/app/org.armagetronad.ArmagetronAdvanced/.org.armagetronad.ArmagetronAdvanced which looks little bad, but it will work correctly and will be safely sandboxed. :-)

Reply:
That would work and would indeed be easier, but there are two problems you can't know about: The killer for me is that the game is available on different platforms and nobody can remember where configuration and data files are supposed to go on each, so we implemented an ingame help page that shows those directories. With --persist, that page is a lie. Well, it's true within the sandbox, but not outside where the user would typically operate.

Less importantly, we moved the configuration files around for 0.4 already (into ~/.config and ~/.local), but would like to be able to import 0.2.9's config, so then we'd need two --persist lines, one just for compatibility. Or some really good forward thinking now for 0.2.9.

I first tried using a wrapper script to use our command line arguments to bend the directories we use, but now I remembered we also have preprocessor defines one can use to hardcode directories. With environment variable substitution to boot! Not my code :) The only difficulty is getting the ${XDG...} construct unevaluated into the code, past JSON and the build scripts.

Reply:
I have suggested a possible solution in: https://github.com/flathub/org.armagetronad.ArmagetronAdvanced/pull/5/commits/43e63fccfa4f29875f372f6106aeac22b86ad998#r548146855 :-)

Reply:
I have suggested a possible solution in: https://github.com/flathub/org.armagetronad.ArmagetronAdvanced/pull/5/files#r548150401 :-)